Added Navitron JS and base plugin CSS

mlworks opened the pull request:
Set up JS functionality and base plugin CSS

Status: **Opened for visibility**

Reviewers: @donnielrt @jeffkamo  
## Changes
- Added JS for setting up and interacting with Navitron
- Added base CSS for plugin
- Set up example page with styles
## How to test-drive this PR
- Copy repo to desktop
- npm install, bower install, grunt serve
- Go to http://localhost:3000

mlworks committed:
Adding Navitron JS and base CSS files

Also quickly whipped up the examples page.

mlworks committed:
Fix jshint error

mlworks committed:
[Examples] Add navitron item selector class

mlworks committed:
[CSS] Add styles for examples page

Commit:
Adding tests

mlworks committed:
DRY'd up some code and add options test

mlworks committed:
[Navitron] Add more tests

mlworks committed:
[Navitron] New mock html

Commit:
Some changes for public methods

Commit:
Merge branch 'testing' of https://github.com/mobify/navitron into testing

bendvc committed:
Corrected the setting of the current pane

mlworks committed:
[Navitron] Initial transform WIP

bendvc committed:
Merge branch 'testing' of https://github.com/mobify/navitron into testing

bendvc committed:
Update ignore file

bendvc committed:
asdf

Commit:
Some refactoring of names etc

Commit:
Merge branch 'testing' of https://github.com/mobify/navitron into testing

Commit:
Added in logic for labelling user passed in markup

mlworks committed:
[Navitron] Build function WIP

Commit:
Updated selector for navitron when labelling tree

mlworks committed:
Merge branch 'testing' of https://github.com/mobify/navitron into testing

Conflicts:
	src/js/navitron.js

mlworks committed:
[Navitron] Refactored example code and build

fixed an event test

mlworks committed:
[Navitron] Fix failing event in tests

Commit:
Consolidating functions

mlworks committed:
[Navitron] Add one more event test

Commit:
Adding hidden attribute to remove FOUC

Commit:
Merge branch 'testing' of https://github.com/mobify/navitron into testing

mlworks committed:
[Navitron] Fix level data logic

and moved some of the init functions into build.

kpeatt commented on dist/navitron.js:
```diff
@@ -0,0 +1,275 @@
+                        // Setting z-index to make sure pane shifting out will always be above
+                        // the pane that's being shifted back in
+                        $currentMenu.css({
+                            zIndex: 2
```
Are you concerned about this hard set z-index? Should this be either relative to an existing element or has this worked fine for you?

mlworks committed:
[Navitron] Enforce one pane animating at a time

mlworks committed:
[Navitron] Fix aria-expanded toggle

mlworks committed:
CSOPS-1371: Fix Android 4.1 paint issues

mlworks committed:
[CSOPS-1371] Reduce box-shadow bug

Can't remove box-shadow bug completely for Legacy Android browsers
unless we use Deckard to scope our CSS. But this solution should be
good where we don't have to rely on Deckard and the pane only have some
rendering artifacts when sliding in/out.

tedtate commented:
@donnielrt @jeffkamo would you guys mind taking a look at this?

jeffkamo commented on src/style/navitron.scss:
```diff
@@ -1,4 +1,85 @@
+//    the next pane over like a page.
+
+.navitron__nested {
+    position: absolute;
```
What is this absolutely positioned to? (I'm assuming `.navitron`, but would be nice to be explicit)

jeffkamo commented on src/style/navitron.scss:
```diff
@@ -1,4 +1,85 @@
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: 10;
```
Why does this need `z-index`?

jeffkamo commented on src/style/navitron.scss:
```diff
@@ -1,4 +1,85 @@
+//
+//    [Note] Must keep this as 'display: none' otherwise you can
+//    crash the browser translating too many divs at once. This
+//    also helps with Pinny's opening/closing animation.
```
Should we be mentioning other plugins in the comments?

jeffkamo commented on src/style/navitron.scss:
```diff
@@ -1,4 +1,85 @@
+
+    transform: translate3d(100%, 0, 0);
+
+    perspective: 1000px; // 2
```
Is there a reason `1000px` was specifically chosen for the value?

jeffkamo commented on src/style/navitron.scss:
```diff
@@ -1,4 +1,85 @@
+// 1. Avoid content bleed through.
+
+.navitron__pane {
+    position: absolute;
```
Same comment as above, it might be nice to explicitly mention what this is absolutely positioned relative to.

jeffkamo replied:
In fact, if this positioning is being applied to both `.navitron__nested` and `.navitron__pane`, would it be worth declaring them once in separate block?

``` scss
.navitron__nested,
.navitron__pane {
    position: absolute;
    // ...
}
```

mlworks committed:
[Navitron] Read me WIP

mlworks committed:
[Navitron] Update Readme + add callback functions

mlworks committed:
[Navitron] More Readme update and JS cleanup

mlworks committed:
[Navitron] Read me fix document styles

mlworks committed:
[Navitron] Update accessibility + showPane method

mlworks committed:
[Navitron] Remove iOS5 from compatibility

mlworks committed:
[Navitron] Add shift demo gif

mlworks committed:
CSOPS-1487: Add ability to use custom markup

for header and footer sections

mlworks committed:
CSOPS-1487: Updating Readme

mlworks committed:
CSOPS-1489: Update docs

mlworks committed:
CSOPS-1487: Add tests for structure option

mlworks committed:
CSOPS-1490: Fix some parsing bugs

mlworks committed:
CSOPS-1490: Ensure next-pane button gets attr

mlworks committed:
CSOPS-1490: 'Add pane' method WIP

mlworks committed:
CSOPS-1490: This method needs rework, removing it

mlworks committed:
Update readme

mlworks committed:
Space out bolded text in Readme

mlworks committed:
CSOPS-1489: Refactored level labelling + functions

renaming for more clarity.

mlworks committed:
CSOPS-1489: Readme typos, rewording for clarity

mlworks committed:
CSOPS-1489: Remove irrelevant text copy

mlworks committed:
CSOPS-1489: Fix methods sub headers to be...

consistent with options.

mlworks committed:
CSOPS-1489: JS/CSS tweaks (PR fixes)

mlworks committed:
CSOPS-1489: More CSS tweaks per PR feedback

mlworks committed:
CSOPS-1489: Readme instructions, keyboard controls

mlworks committed:
More keyboard functionality tweaks

mlworks committed:
CSOPS-1565: Keyboard controls

mlworks committed:
CSOPS-1565: Finish keyboard controls

mlworks committed:
Issue #4: Remove gitignore 'dist' folder

mlworks committed:
Issue #2: Structure option consistent with Pinny

mlworks committed:
Issue #3: Sibling <ul> now grouped in same pane

mlworks committed:
Issue #3: Allow pane to group contents beyond <ul>

mlworks committed:
Grab all contents including text nodes

mlworks committed:
Merge branch 'testing' into development

mlworks committed:
Fix adjacent <ul> wasn't grouped together

donnielrt commented:
Hey, we’re looking to prune older unattended PRs. If this PR is still relevant and you would like to see it merged in, please reopen the PR and we’ll add it to our backlog! Thanks!
